samples/scala/spark/sources/SVD.scala

```diff
@@ -102,100 +102,78 @@ object SVD {
                                                           Matrices.dense(nValues, nValues, V))
     }
 
-    private def getAsPairRDD(matrix: RowMatrix): RDD[Tuple2[HomogenNumericTable, Long]] = {
-        val nCols = matrix.numCols.toInt
+
+    private def getAsPairRDD(matrix: RowMatrix): RDD[HomogenNumericTable] = {
+
 
         val tablesRDD = matrix.rows.mapPartitions(
             (it: Iterator[Vector]) => {
-                val maxRows: Int = 200000
-                var curRow: Int = 0
+
                 val tables = new ArrayBuffer[HomogenNumericTable]()
                 var arrays = new ArrayBuffer[Array[Double]]()
-                var hasNext = it.hasNext
-                while (hasNext) {
-                    val curVector = it.next
-                    hasNext = it.hasNext
+                it.foreach{curVector =>
                     val rowData = curVector.toArray
                     arrays += rowData
+                }
+                val numCols: Int = arrays(0).length
+                val numRows: Int = arrays.length
 
-                    curRow += 1
-                    if (curRow == maxRows || !hasNext) {
-                        val arrData = new Array[Double](curRow * nCols)
-                        var i: Int = 0
-                        for (i <- 0 to (curRow - 1)) {
-                            arrays(i).copyToArray(arrData, i * nCols)
-                        }
-                        val context = new DaalContext()
-                        val table = new HomogenNumericTable(context, arrData, nCols, curRow)
-
-                        table.pack()
-                        tables += table
-
-                        context.dispose()
-                        curRow = 0
-                        arrays.clear
-                    }
+                val arrData = new Array[Double](numRows * numCols)
+                var i: Int = 0
+                for (i <- 0 to (numRows - 1)) {
+                    arrays(i).copyToArray(arrData, i * numCols)
                 }
+                val context = new DaalContext()
+                val table = new HomogenNumericTable(context, arrData, numCols, numRows)
+                table.pack()
+                tables += table
+
+                context.dispose()
+                arrays.clear
                 tables.iterator
             }
-        )
-
-        tablesRDD.zipWithIndex.cache
+        ).persist(StorageLevel.MEMORY_AND_DISK)
```

Comment:
Why exactly this storage level is used? Is it optimal for any dimensions?

Reply:
This storage level is used, because in case, when data doesn't fit into memory, it will be written to disk. The memory_only level, that used before, recomputes some partitions if they don't fit into memory, which may be inefficient.

Reply:
On small dimensions, this storage level = Memory_only
At large dimensions, those partitions that don't fit into memory will be written to disk. Those partisans that did not fit in memory on disk will be recomputed.